### PR TITLE
[FLINK-11872][table-runtime-blink] update lz4 license file.

### PR DESCRIPTION
**Pull request by KurtYoung**
From my understanding, we should only add license to maven artifacts if the project is also apache 2 license, right? @twalthr

### CODE REVIEW

**Comment by twalthr** [comment]
@flinkbot approve-until architecture
@flinkbot attention @zentol

**Comment by zentol** [comment]
@flinkbot disapprove consensus

**Comment by KurtYoung** [comment]
Thanks @zentol for pointing this out. I was wondering how can i tell whether a dependent jar is packaged in flink's jar?

**Comment by zentol** [comment]
Check the `pom.xml` of the respective module X. There must be an explicit shade/jar/assembly-plugin configuration for anything to be included in the jar. Repeat this process for all modules that depend on X.

**Comment by KurtYoung** [comment]
I see, thanks @zentol .
@twalthr , do i have to shade lz4-java into `table-runtime-blink` package? What is the standard to decide whether to shade a dependency or not?

**Comment by twalthr** [comment]
@KurtYoung Currently, we shade everything for the `flink-table` package except for Calcite and Janino. Calcite could not be shaded because it was exposed through the API but this might change soon. Janino had some problems with a unshaded properties file. There is no explicit rule of shading everything but every shaded dependency is a potential dependency conflict less for users.

**Comment by KurtYoung** [comment]
Thanks @twalthr , since lz4 is widely used in other projects, i would shade it into `flink-table`.

**Comment by KurtYoung** [comment]
@zentol could you take a look again?

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/compression/Lz4BlockCompressionFactory.java [hunk 1]
```diff
@@ -23,7 +23,6 @@
  */
 public class Lz4BlockCompressionFactory implements BlockCompressionFactory {
 
-
```
Comment (zentol) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/compression/Lz4BlockCompressionFactory.java]: unrelated

File: flink-table/flink-table-runtime-blink/pom.xml [hunk 2]
```diff
@@ -85,4 +85,28 @@ under the License.
...
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>shade-flink</id>
+						<configuration>
+
```
Comment (zentol) [inline flink-table/flink-table-runtime-blink/pom.xml]: let's remove this empty line

File: flink-table/flink-table-runtime-blink/pom.xml [hunk 3]
```diff
@@ -85,4 +85,28 @@ under the License.
...
+							<artifactSet>
+								<includes combine.children="append">
+									<!-- lz4-java -->
```
Comment (zentol) [inline flink-table/flink-table-runtime-blink/pom.xml]: unnecessary

**Comment by KurtYoung** [comment]
@zentol updated
